src/Plugin/Action/EmitFileEvent.php

```diff
@@ -2,17 +2,11 @@
 
 namespace Drupal\islandora\Plugin\Action;
 
+use Drupal\islandora\EventGenerator\EmitEvent;
+
 use Drupal\Core\Entity\EntityInterface;
-use Drupal\Core\Entity\EntityTypeManagerInterface;
 use Drupal\Core\File\FileSystemInterface;
-use Drupal\Core\Messenger\MessengerInterface;
-use Drupal\Core\Session\AccountInterface;
 use Drupal\Core\Site\Settings;
-use Drupal\Core\StreamWrapper\StreamWrapperManager;
```

Comment:
`StreamWrapperManager` is still used on line 109/53, I tried to add an audio media and got a WSOD. 

Stacktrace in the apache log refers to  
```
... Error: Class 'Drupal\\islandora\\Plugin\\Action\\StreamWrapperManager' not found in /home/vagrant/all_claw/islandora/src/Plugin/Action/EmitFileEvent.php ...
```

Reply:
Ah... stuff was originally developed against (a fork of) 1.1.1... which was still using [the (now deprecated) `$fileSystem->uriScheme()` thing](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21File%21FileSystemInterface.php/function/FileSystemInterface%3A%3AuriScheme/8.9.x)... patch applied relatively cleanly (IIRC), and the actions were giving a checkmark... I guess this class isn't actually covered under any (unit) tests?

Anyway... the `$fileSystem` being injected here is no longer being used. Should we drop it? Would rather simplify the class. [Removal should be inline with Drupal's policies WRT protected properties?](https://www.drupal.org/about/core/policies/core-change-policies/drupal-8-and-9-backwards-compatibility-and-internal-api#protected-properties):

> Protected properties on a class are always considered `@internal` unless they're on a base class marked with `@api`

Reply:
Probably no test, if you want to add a unit test that would be 🥇 . As for the `$fileSystem`, you are correct that it does appear to not be necessary, I'd vote to get rid of it and simplify.

Reply:
Rebased on top of `HEAD`, and added the reference here back in... left `$fileSystem` (for now?).

Reply:
_Did_ take a look at how tests might be written for this `::generateData()` stuff; however, the method is `protected`... would get into either changing the visibility (and pushing such a change up into the interface and propagating around), using reflection to adjust the visibility when running the test (which is relatively simple but gross) or mocking/loading up all the services required to `::execute()` the event, which... is... rather more complex.

... start wondering if things should be structured differently, to facilitate testing?... anyway...